### PR TITLE
fix(color-contrast): ignore zero width characters

### PR DESCRIPTION
**Pull request by straker**
I grabbed the Affirm font from the provided website. I couldn't figure out how to generate a font that had 0 width characters so I hope it's OK for us to grab it for test purposes. 

Closes: #3918

### CODE REVIEW

File: lib/commons/text/is-icon-ligature.js [hunk 1]
```diff
@@ -143,6 +139,14 @@ export default function isIconLigature(
   const firstChar = nodeValue.charAt(0);
   let width = canvasContext.measureText(firstChar).width;
 
+  // if the font uses zero-width characters that are not unicode characters
+  // it has to be an icon ligature (e.g. Affirm)
+  // @see https://github.com/dequelabs/axe-core/issues/3918
+  if (width === 0) {
+    font.numLigatures++;
+    return true;
```
Comment (WilcoFiers) [inline lib/commons/text/is-icon-ligature.js:147]: Why do you think a zero-width char is a ligature font? Unicode has at least one character that is supposed to have zero width (the zero-width space). There might be others, I'm not particularly well versed on unicode. This seems like a false assumption. Would it be better to loop over characters until you found one that has space?

File: test/assets/Affirm.woff
Comment (WilcoFiers) [inline test/assets/Affirm.woff:1]: Are we allowed to copy this into our repo?

**Review by WilcoFiers — DISMISSED** [review]

File: lib/commons/text/is-icon-ligature.js (hunk 1 above)
Comment (straker) [reply lib/commons/text/is-icon-ligature.js:147]: https://github.com/dequelabs/axe-core/pull/4102 will solve the zero-width unicode problem

File: test/assets/Affirm.woff
Comment (dbjorge) [reply test/assets/Affirm.woff:1]: I went ahead and replaced it with a purpose-built font that just has a zero-width `0` character and nothing else

File: lib/commons/text/is-icon-ligature.js [hunk 2]
```diff
@@ -143,6 +139,14 @@ export default function isIconLigature(
   const firstChar = nodeValue.charAt(0);
   let width = canvasContext.measureText(firstChar).width;
 
+  // if the font uses zero-width characters that are not unicode characters
+  // it has to be an icon ligature (e.g. Affirm)
```
Comment (dbjorge) [inline lib/commons/text/is-icon-ligature.js:143]:
```suggestion
  // we already checked for typical zero-width unicode formatting characters further up,
  // so we assume that any remaining zero-width characters are part of an icon ligature
```

**Review by dbjorge — APPROVED** [review]
Updated inline with a non-copyright-encumbered custom font per our discussion offline. Minor suggestion on one of the new comments, otherwise LGTM!

**Review by dbjorge — APPROVED** [review]

**Pull request merged**